File: pdu/python/hako_msgs/pdu_cdr_conv_SimTime.py
from .pdu_pytype_SimTime import SimTime
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_SimTime(writer: CdrWriter, src: SimTime):
    writer.write_uint64(src.time_usec)


def cdr_body_to_py_SimTime(reader: CdrReader, dst: SimTime):
    dst.time_usec = reader.read_uint64()
    return dst


def py_to_cdr_SimTime(src: SimTime) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_SimTime(writer, src)
    return writer.bytes()


def cdr_to_py_SimTime(cdr_payload) -> SimTime:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = SimTime()
    return cdr_body_to_py_SimTime(reader, dst)

File: pdu/javascript/hako_msgs/pdu_cdr_conv_SimTime.js
import { SimTime } from './pdu_jstype_SimTime.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduSimTimeConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { SimTime } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint64(src.time_usec);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { SimTime } dst
     * @returns { SimTime }
     */
    static cdr_body_to_js(reader, dst) {
        dst.time_usec = reader.read_uint64();
        return dst;
    }

    /**
     * @param { SimTime } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { SimTime }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new SimTime());
    }
}
